src/components/sketches/Scene5_happy.ts

```diff
@@ -0,0 +1,266 @@
+      this.rand = p.random(0, 3);
+      this.vel = this.isFirework
+        ? p.createVector(0, p.random(-12, -8))
+        : p5.Vector.random2D().mult(p.random(2, 10));
+    }
+
+    update(force: p5.Vector) {
+      if (!this.isFirework) {
+        this.vel.mult(0.9);
+        this.lifespan -= 4;
+      }
+      this._acc.add(force);
+      this.vel.add(this._acc);
+      this.pos.add(this.vel);
+      this._acc.mult(0);
+    }
+
+    done() {
+      return this.lifespan < 0;
+    }
+
+    show() {
+      if (!this.isFirework) {
+        // after explosion
+        p.strokeWeight(6);
+        p.stroke(255, 211, 97, this.lifespan);
+      } else {
+        p.strokeWeight(8);
+      }
+
+      const opacity = this.rand < 1 ? 0.2 : this.rand < 2 ? 0.4 : 0.6;
+      p.stroke(`rgba(255,211,97,${opacity})`);
+      p.point(this.pos.x, this.pos.y);
+    }
+  }
+
+  class RingParticle {
+    private _pos: p5.Vector;
+    private _vel: p5.Vector;
+    private _acc: p5.Vector;
+    private _radius: number;
+    private _time: number;
+    private _explode: boolean;
+    private _lifespan: number;
+    private _history: p5.Vector[];
+
+    constructor(x: number, y: number, theta: number) {
+      this._pos = p.createVector(x, y); // Position
+      this._vel = p.createVector(2, 2); // Velocity
+      this._radius = RING_RADIUS; //3*noise(this.pos.x/250, this.pos.y/250)*5;
+      this._vel.limit(5);
+      this._acc = p.createVector(0.1, 0.1); // Acceleration
+      this._vel.rotate(theta); // Rotate velocity to the given rotation
+      this._acc.rotate(theta); // Rotate to the given rotation
+      this._time = 0; // Time alive
+      this._explode = false;
+
+      this._lifespan = FIREWORK_LIFESPAN;
+      this._history = [];
+    }
+    public update() {
+      this._vel.add(this._acc);
+      this._pos.add(this._vel); // Update position with velocity
+      // If older than 40, be deleted next draw event
+      if (this._time <= 40) {
+        // If older than 15, reduce the size of the ringParticles every step
+        if (this._time > 15) {
+          this._radius -= 0.05;
+        }
+        this._time++;
+      } else {
+        if (this._lifespan == FIREWORK_LIFESPAN) {
+          this._explode = true;
+          this._acc.add(p.createVector(0, 0.3));
+
+          this._vel = p5.Vector.random2D();
+          this._vel.mult(p.random(2, 10));
+        }
+        this._vel.mult(VELOCITY_CHANGE); // vel change ***
+        this._lifespan -= LIFESPAN_DECREMENT; // lifespan ***
+
+        const v = p.createVector(this._pos.x, this._pos.y);
+        this._history.push(v);
+
+        //size of trail ***
+        if (this._history.length > TRAIL_SIZE) {
+          this._history.shift();
+        }
+      }
+    }
+    public delete() {
+      return this._lifespan < 0;
+    }
+
+    public show() {
+      if (this._explode == false) {
+        p.noStroke();
+        // Set colour based on average of x and y position
+        p.fill(255, 211, 97, 204);
```

Comment:
im confused.. how is this the average of x and y?

Reply:
poor comment, this was when it was mutlicolored, but we are sticking with yellow

Reply:
deleted this comment